src/features/reviews/utils/adjustQueue.js

```diff
@@ -9,15 +9,35 @@
 
 import _ from 'lodash';
 
-import { MEANING, RADICAL } from 'src/common/constants';
+import { MEANING, RADICAL, RANDOM_ORDER, LOWEST_LEVEL_FIRST, CURRENT_LEVEL_FIRST, ASCENDING_SRS_STAGE, DESCENDING_SRS_STAGE } from 'src/common/constants';
 
 const ALLOWED_MAX_DISTANCE = 10;
 const ALLOWED_MIN_DISTANCE = 3;
 
-export default (queue, backToBackMode = false, meaningFirst = false) => {
+export default (queue, backToBackMode = false, meaningFirst = false, reviewOrder = RANDOM_ORDER) => {
+
   const newQueue = queue.slice();
   let currentIndex = 0;
 
+  // sort new queue based on the selected review order
+  switch (reviewOrder) {
+    case RANDOM_ORDER:
+      newQueue.sort(() => Math.random() - 0.5);
+      break;
+    case LOWEST_LEVEL_FIRST:
+      newQueue.sort((a, b) => a.subjectLevel - b.subjectLevel);
+      break;
+    case CURRENT_LEVEL_FIRST:
+      newQueue.sort((a, b) => b.subjectLevel - a.subjectLevel);
+      break;
+    case ASCENDING_SRS_STAGE:
+      newQueue.sort((a, b) => a.review.data.srs_stage - b.review.data.srs_stage);
```

Comment:
I'm a little concerned that if `a.review` or `a.review.data` was `undefined` / `null` the app would crash. I wonder if there's a chance for that to happen 🤔 Maybe we could take a defensive approach here and use lodash `get` method for this, maybe something like:

```javascript
newQueue.sort((a, b) => _.get(a, 'review.data.srs_stage') - _.get(b, 'review.data.srs_stage'));
```

Reply:
Since you mentioned `loadsh` for the shuffling, browsing through the documentation I found the (`_.orderBy`)[https://lodash.com/docs/4.17.15#orderBy] function that fits like a glove here, so I refactored this part to use `lodash` for all the `Review Order` implementations.